DeviceSDK/src/main/java/org/astarteplatform/devicesdk/AstarteDeviceIdUtils.java

```diff
@@ -0,0 +1,36 @@
+package org.astarteplatform.devicesdk;
+
+import java.nio.ByteBuffer;
+import java.util.Base64;
+import java.util.UUID;
+import org.astarteplatform.devicesdk.util.UUID5;
+
+/** The class helps to generate deviceId for Astarte Device API */
+public class AstarteDeviceIdUtils {
+
+  /**
+   * Generate a device Id in a random way.
```

Comment:
I think that it is important to tell our users that it is an UUIDv4 string. We might also document that the string is base64 encoding of UUID bytes.